Mathlib/LinearAlgebra/Matrix/ToLin.lean

```diff
@@ -1008,3 +1008,39 @@ def algEquivMatrix [Fintype n] (h : Basis n R M) : Module.End R M ≃ₐ[R] Matr
 #align alg_equiv_matrix algEquivMatrix
 
 end
+
+namespace Basis
+
+variable {R M ι : Type*} [CommRing R] [AddCommGroup M] [Module R M] [Fintype ι] [DecidableEq ι]
+
+/-- The standard basis of the endomorphism algebra of a module
+induced by a basis of the module.
+
+If `M` is a module with basis `b` indexed by a finite type `ι`,
+then `Basis.end b` is the basis of `Module.End R M` indexed by `ι × ι`
+where `(i, j)` indexes the linear map that sends `b j` to `b i`
+and sends all other basis vectors to `0`.  -/
+@[simps! repr_apply]
+noncomputable
+def _root_.Basis.end (b : Basis ι R M) : Basis (ι × ι) R (Module.End R M) :=
+  (Matrix.stdBasis R ι ι).map (LinearMap.toMatrix b b).symm
+
+-- the left hand side simplifies, so this is a bad simp lemma
```

Comment:
```suggestion
```

Reply:
I added this comment to explain why I do not let `simps` generate the lemma: the linter complains.